wiki/mapping/map-format.md

```diff
@@ -313,16 +313,456 @@ The exact specifics of what goes in `_customData` is entirely dependent on commu
+While `d` can go into negative numbers, be aware that this has some unintended effects.
+
+#### w
+An integer which which represents the width of the obstacle.
+While `w` can go into negative numbers, be aware that this has some unintended effects.
+
+#### h
+An integer numbers from 1 to 5, which represents the height of the obstacle.
+While `h` can go into negative numbers, be aware that this has some unintended effects.
+
+### Sliders
+These describe arcs. If the head or tail of an arc matches a note's time and position,
+then the arc will connect with the note and alter how the note is scored.
+
+```json
+{
+  "b": 10.0,  // Head Beat
+  "c": 0,     // Color
+  "x": 1,     // Head x
+  "y": 0,     // Head y
+  "d": 1,     // Head direction
+  "mu": 1.0,  // Head multiplier
+  "tb": 15.0, // Tail Beat
+  "tx": 2,    // Tail x
+  "ty": 2,    // Tail y
+  "tc": 1,    // Tail direction
+  "tmu": 1.0, // Tail Multiplier
+  "m": 1,     // Mid-anchor mode
+}
+```
+
+#### b
+The time, in beats, where this head of this object reaches the player.
+
+#### c
+An integer which represents the color of the note.
+|`_c`|Result|
+|:---:|-----|
+|`0`|Red|
+|`1`|Blue|
+
+#### x
+An integer number, from 0 to 3, which represents the column where the head of the arc is located.
+The far left column is located at index 0, and increases to the far right column located at index 3.
+
+#### y
+An integer number, from 0 to 2, which represents the layer where the head of the arc is located.
+The bottommost layer is located at layer 0, and inceases to the topmost layer located at index 2.
+
+#### d
+An integer number which represents the head direction of the arc.
+
+|`_d`|Result|
+|:-------------------:|-------------------|
+|`0`|Up|
+|`1`|Down|
+|`2`|Left|
+|`3`|Right|
+|`4`|Up Left|
+|`5`|Up Right|
+|`6`|Down Left|
+|`7`|Down Right|
+|`8`|Any (Dot Note)|
+
+#### mu
+A float which represents how far the arc goes from the head of the arc.
+
+#### tb
+The time, in beats, where this tail of this object reaches the player.
+
+#### tx
+An integer number, from 0 to 3, which represents the column where the tail of the arc is located.
+The far left column is located at index 0, and increases to the far right column located at index 3.
+
+#### ty
+An integer number, from 0 to 2, which represents the layer where the tail of the arc is located.
+The bottommost layer is located at layer 0, and inceases to the topmost layer located at index 2.
+
+#### tc
+An integer number which represents the tail direction of the arc.
+
+|`_d`|Result|
+|:-------------------:|-------------------|
+|`0`|Up|
+|`1`|Down|
+|`2`|Left|
+|`3`|Right|
+|`4`|Up Left|
+|`5`|Up Right|
+|`6`|Down Left|
+|`7`|Down Right|
+|`8`|~~Any (Dot Note).~~ Functions as Down.|
+
+#### tmu
+A float which represents how far the arc goes from the tail of the arc.
+
+#### m
+An integer number which represents how the arc curves from the head to the mid point of the arc.
```

Comment:
Mid anchor mode only applies in certain situations, which should probably be mentioned here. It applies when 2 conditions are met:
- Head and tail line index are equal
- Head and tail cut directions are equal OR their angle difference is 180

Notably the angle difference is NOT an absolute value, which means that only half of the opposing direction pairs will meet this condition. These pairs are (head -> tail):
- Down -> Up
- Right -> Left
- DownRight -> UpLeft
- UpRight -> DownLeft

This honestly feels like an oversight in the game, so it may get changed in the future.